Wyrównanie nazw lekcji względem siebie

Kazoroo opened the pull request:
Jeśli godzina jest mniejsza od 10 dodaje się spacja na początku całej godziny i na jej końcu, przez co nazwy lekcji są wyrównane względem siebie. Rozwiązuje issue [#184](https://github.com/szkolny-eu/szkolny-android/issues/184).

![Zrzut ekranu 2024-04-11 204919](https://github.com/szkolny-eu/szkolny-android/assets/91783342/e9253cf0-de9b-4e90-a4cd-4b897ff41688)

kuba2k2 reviewed (changes requested):
Nie można zmieniać tego co zwraca model Time, bo czas w takiej postaci zostanie zapisany do bazy danych. Prędzej czy później spowoduje to problemy lub crashe aplikacji.

Należy ten warunek dodający spację umieścić w kodzie karty HomeTimetableCard, w miejscu gdzie Time przetwarzany jest na string oraz dodawany do reszty lekcji.  

Dodatkowo, co wprowadza zmiana w XMLu w tym commicie?

Kazoroo commented:
Dodałem funkcję w HomeTimetableCard która dodaje spacje, a zmiany w Time usunąłem i zostawiłem takie jakie było. Zmiany w XML zostały wprowadzone omyłkowo, i w żaden sposób nie zmieniają wyglądu aplikacji.

kuba2k2 commented on app/src/main/java/pl/szczodrzynski/edziennik/ui/home/cards/HomeTimetableCard.kt:
```diff
@@ -343,6 +343,19 @@ class HomeTimetableCard(
         b.nextLessons.text = text.concat("\n")
     }}
 
+    private fun adjustTimeWidth(time: String?): String {
+        if (time != null) {
+            //compared to 4 because when hour is less than 10 there are 4 chars (hour, colon and minutes)
+            if(time.length == 4) {
+                return " $time  "
+            } else {
+                return "$time "
+            }
+        } else {
+            return ""
+        }
+    }
```
This can be replaced with a `when` block and the function converted to an expression body, like this:
```suggestion
    private fun adjustTimeWidth(time: String?) = when {
        time == null -> ""
        // compared to 4 because when hour is less than 10 there are 4 chars (hour, colon and minutes)
        time.length == 4 -> " $time "
        else -> "$time "
    }
```

Please check if this works, because I can't do that right now.

kuba2k2 commented on app/src/main/res/layout/material_drawer_header.xml:
```diff
@@ -68,9 +68,9 @@
         android:visibility="visible"
-        app:materialDrawerSelectorOnPress="#80ffffff"
         app:layout_constraintEnd_toStartOf="@id/material_drawer_account_header_small_second"
-        app:layout_constraintTop_toBottomOf="@+id/material_drawer_statusbar_guideline" />
+        app:layout_constraintTop_toBottomOf="@+id/material_drawer_statusbar_guideline"
+        app:materialDrawerSelectorOnPress="#80ffffff" />
```
If these changes are unintentional, please revert them.

Kazoroo commented:
Okay, I struggled a bit with git, but I reverted the wrong XML code. Now it is as before. Your suggestion to use when statement is good, I've implemented this solution. So now the only changes to the code are in the HomeTimetableCard.kt file.
